packages/next/src/server/lib/start-server.ts

```diff
@@ -234,9 +237,12 @@ export async function startServer(
 
       port = typeof addr === 'object' ? addr?.port || port : port
 
-      const networkUrl = hostname
-        ? `${selfSignedCertificate ? 'https' : 'http'}://${actualHostname}:${port}`
+      const networkHostname =
+        hostname ?? getNetworkHost(isIPv6(actualHostname) ? 'IPv6' : 'IPv4')
+      const networkUrl = networkHostname
+        ? `http://${formatHostname(networkHostname)}:${port}`
```

Comment:
this should have the same logic as below for determining the protocol right?

Reply:
Oh yea good catch